plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerTableStatistics.java

```diff
@@ -0,0 +1,440 @@
+        assertUpdate("DROP TABLE IF EXISTS " + tableName);
+        computeActual("" +
+                "CREATE TABLE " + tableName + " AS SELECT " +
+                "  orderkey, " +
+                "  'abc' v3_in_3, " +
+                "  CAST('abc' AS varchar(42)) v3_in_42, " +
+                "  if(orderkey = 1, '0123456789', NULL) single_10v_value, " +
+                "  if(orderkey % 2 = 0, '0123456789', NULL) half_10v_value, " +
+                "  if(orderkey % 2 = 0, CAST((1000000 - orderkey) * (1000000 - orderkey) AS varchar(20)), NULL) half_distinct_20v_value, " + // 12 chars each
+                "  CAST(NULL AS varchar(10)) all_nulls " +
+                "FROM tpch.tiny.orders " +
+                "ORDER BY orderkey LIMIT 100");
+        try {
+            gatherStats(tableName);
+            assertQuery(
+                    "SHOW STATS FOR " + tableName,
+                    "VALUES " +
+                            "('orderkey', null, 100, 0, null, null, null)," +
+                            "('v3_in_3', 600, 1, 0, null, null, null)," +
+                            "('v3_in_42', 600, 1, 0, null, null, null)," +
+                            "('single_10v_value', 20, 1, 0.99, null, null, null)," +
+                            "('half_10v_value', 1000, 1, 0.5, null, null, null)," +
+                            "('half_distinct_20v_value', 1200, 50, 0.5, null, null, null)," +
+                            "('all_nulls', 0, 0, 1, null, null, null)," +
+                            "(null, null, null, null, 100, null, null)");
+        }
+        finally {
+            assertUpdate("DROP TABLE " + tableName);
+        }
+    }
+
+    @Override
+    @Test
+    public void testPartitionedTable()
+    {
+        throw new SkipException("Not implemented"); // TODO
+    }
+
+    @Override
+    @Test
+    public void testView()
+    {
+        String tableName = "test_stats_view";
+        sqlServer.execute("CREATE VIEW " + tableName + " AS SELECT orderkey, custkey, orderpriority, comment FROM orders");
+        try {
+            assertQuery(
+                    "SHOW STATS FOR " + tableName,
+                    "VALUES " +
+                            "('orderkey', null, null, null, null, null, null)," +
+                            "('custkey', null, null, null, null, null, null)," +
+                            "('orderpriority', null, null, null, null, null, null)," +
+                            "('comment', null, null, null, null, null, null)," +
+                            "(null, null, null, null, null, null, null)");
+            // It's not possible to ANALYZE a VIEW in SQL Server
+        }
+        finally {
+            sqlServer.execute("DROP VIEW " + tableName);
+        }
+    }
+
+    @Override
+    public void testMaterializedView()
+    {
+        throw new SkipException("see testIndexedView");
+    }
+
+    @Test
+    public void testIndexedView() // materialized view
```

Comment:
Why not move this to `testMaterializedView`?

Reply:
"indexed view" is how it's called in SQL Server world, so it's what you would want to find in test code, if you wanted to find a test for an indexed view

Reply:
Fair point.